### ISSUE TITLE
Switch latency traking in DS stats to System.nanoTime()

### ISSUE THREAD

**Issue by puzpuzpuz** [hazelcast/hazelcast#16767]
@alex-dukhno found that we use `System.currentTimeMillis()` for latency tracking in maps: https://github.com/hazelcast/hazelcast/blob/40cadaef1f4e944d17f77a220704e964c9af6351/hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/AbstractReplicatedRecordStore.java#L195

So, under certain conditions we may have zero or even negative latency values. We should start using `System.nanoTime()` (for latency tracking, at least) which has better granularity and monotonicity guarantees. That could be implemented by changing default implementation of `com.hazelcast.internal.util.Clock`, but such change should be made very cautiously to avoid regressions in other modules that consume `Clock` implementations.

cc @emre-aydin @mmedenjak

**Comment by puzpuzpuz** [comment]
cc @pveentjer

**Comment by pveentjer** [comment]
Tracking time with a milisecond granularity sn't that useful because we can't track call that have submillisecond latencies.

So in theory I have no problem making use of nanotime; but it needs to be done consistently because nanotime should be used very carefully. You don't want to run into overflow problems for example.

**Comment by puzpuzpuz** [comment]
> Tracking time with a milisecond granularity sn't that useful because we can't track call that have submillisecond latencies.

I agree. Once we have histogram proves for latencies, it will make much more sense.

> So in theory I have no problem making use of nanotime; but it needs to be done consistently because nanotime should be used very carefully. You don't want to run into overflow problems for example.

Good point. Overflow may be a problem indeed, as a `long` allows to hold ~292 years with nanosecond granularity.

In case of total latency counter probes, users may experience the overflow on members running for several years (or even months) as operations are executed in parallel and latencies for neighbor operations may intersect in terms of time. So, maybe we should change granularity to micros or millis in these places:
* https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/LocalMapStatsImpl.java#L99-L104
* https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/LocalReplicatedMapStatsImpl.java#L78-L89

@alex-dukhno @blazember WDYT?
